src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -579,7 +579,7 @@ private Conversion ClassifyExplicitBuiltInOnlyConversion(TypeSymbol source, Type
 
             if (HasUnboxingConversion(source, destination, ref useSiteDiagnostics))
             {
-                return Conversion.Unboxing;
+                return source.Kind == SymbolKind.TypeParameter ? Conversion.TypeParameterUnboxing : Conversion.ValueTypeUnboxing;
```

Comment:
>return source.Kind == SymbolKind.TypeParameter ? Conversion.TypeParameterUnboxing : Conversion.ValueTypeUnboxing; [](start = 16, length = 113)

It looks like VB classifies a bunch of conversions to type parameter as a TypeParameter conversion.